src/test/ui/deref-suggestion.stderr

```diff
@@ -0,0 +1,32 @@
+error[E0308]: mismatched types
+  --> $DIR/deref-suggestion.rs:16:9
+   |
+16 |     foo(r_s);
+   |         ^^^ expected struct `std::string::String`, found reference
+   |
+   = note: expected type `std::string::String`
+              found type `&std::string::String`
+   = help: try with `*r_s`
```

Comment:
The suggestion is still wrong. I think this should only trigger for expressions that produce a temporary or `Copy` types. Everything else should be handled by the method suggesting code that gets triggered on e.g. `let u: usize = "";` producing a suggesting for `.len()`.